mqt/syrec/syrec_editor.py

```diff
@@ -193,40 +237,51 @@ def load(self, filename):
         return
 
     def build(self):
+
         if self.before_build is not None:
             self.before_build()
 
         self.writeEditorContentsToFile()
 
-        prog = program()
+        if self.buildType== [0,0]:
 
-        error_string = prog.read("/tmp/out.src")
-
-        if error_string == "PARSE_STRING_FAILED":
             if self.parser_failed is not None:
-                self.parser_failed("Editor is Empty")
+                self.parser_failed("Sythesizer not selected")
             return
 
-        elif error_string != "":
-            if self.build_failed is not None:
-                self.build_failed(error_string)
-            return
+        else:
```

Comment:
this indentation change is not necessary as all previous if statements contain a `return` statement. Just remove the `else:` and keep the previous indentation.